### PR TITLE
build llvm for rill-head by Dockerfile

### PR DESCRIPTION
**Pull request by yutopp**
Build LLVM for `rill-head` by Dockerfile instead of install.sh. This change will reduce building time :tada:

### CODE REVIEW

File: build/rill-head/docker/Dockerfile [hunk 1]
```diff
@@ -7,9 +7,34 @@ RUN apt-get update && \
...
+          -DLLVM_INCLUDE_TESTS=OFF \
+          ..
+RUN cd llvm-3.9.1.src/build && \
+    cmake --build . --target package/fast -- -j2 && \
+    cmake --build . --target llc/fast -- -j2 && \
+    cmake --build . --target llvm-config/fast -- -j2
+RUN cd llvm-3.9.1.src/build && \
+    cmake --build . --target install && \
+    cp bin/llvm-config /opt/wandbox/rill-head/bin/. && \
+    cp bin/llc /opt/wandbox/rill-head/bin/.
```
Comment (melpon) [inline build/rill-head/docker/Dockerfile]: Don't use mounted volume `/opt/wandbox`.

File: build/rill-head/docker/Dockerfile [hunk 2]
```diff
@@ -7,9 +7,34 @@ RUN apt-get update && \
     apt-get clean -qq -y && \
     rm -rf /var/lib/apt/lists/*
 
+RUN wget -q http://www.llvm.org/releases/3.9.1/llvm-3.9.1.src.tar.xz -O - | tar -xJ && \
```
Comment (melpon) [inline build/rill-head/docker/Dockerfile]: `http` is insecure. Check binary hash like [this](https://github.com/melpon/melpon.org-builder/blob/master/build/clang/docker/Dockerfile#L28) or use `https` by *trusted certificate*.

File: build/rill-head/docker/Dockerfile (hunk 1 above)
Comment (yutopp) [reply build/rill-head/docker/Dockerfile]: Fixed.

File: build/rill-head/docker/Dockerfile (hunk 2 above)
Comment (yutopp) [reply build/rill-head/docker/Dockerfile]: Fixed.

File: build/rill-head/docker/Dockerfile [hunk 3]
```diff
@@ -7,9 +7,39 @@ RUN apt-get update && \
...
+    mkdir build && \
+    cd build && \
+    cmake -G 'Unix Makefiles' \
+          -DCMAKE_INSTALL_PREFIX=/opt/llvm \
+          -DCMAKE_BUILD_TYPE=Release \
+          -DLLVM_TARGETS_TO_BUILD="X86" \
+          -DLLVM_BUILD_TOOLS=OFF \
+          -DLLVM_INCLUDE_EXAMPLES=OFF \
+          -DLLVM_INCLUDE_TESTS=OFF \
+          ..
```
Comment (melpon) [inline build/rill-head/docker/Dockerfile:26]: Merge above two layers and remove `llvm-3.9.1.src.tar.xz` for [minimum image](http://qiita.com/g_ryotaro/items/10c0017671d0ea2635dd).

Comment (yutopp) [reply build/rill-head/docker/Dockerfile:26]: Done!

**Comment by melpon** [comment]
Thank you for your progress!

**Comment by yutopp** [comment]
Thank you for taking your time to the review and quick response :bowing_man:

**Pull request merged**